opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -350,35 +445,40 @@ def get_meter(
     return meter_provider.get_meter(name, version)
 
 
+def _set_meter_provider(meter_provider: MeterProvider, *, log: bool) -> None:
+    def set_mp() -> None:
+        global _METER_PROVIDER  # pylint: disable=global-statement
+        _METER_PROVIDER = meter_provider
+
+        # gives all proxies real instruments off the newly set meter provider
+        _PROXY_METER_PROVIDER.set_meter_provider(meter_provider)
+
+    did_set = _METER_PROVIDER_SET_ONCE.do_once(set_mp)
```

Comment:
This is broken in the Trace API. There is no protection against multiple threads trying to set the MeterProvider at once.

Reply:
https://github.com/open-telemetry/opentelemetry-python/pull/2182 fixes it